lib/triangle_cases.rb

```diff
@@ -1,28 +1,34 @@
 class TriangleCase < OpenStruct
-  def name
+  def test_name
     initial = description.downcase
     replaced = initial.gsub(/(true|false)/, replacements)
     if initial.eql?(replaced) && !initial.include?(triangle)
       replaced = triangle + ' triangle ' + initial
     end
-    'test_%s' % replaced.gsub(/[ ,-]/, '_')
+    'test_%s' % replaced.gsub(/[, -]/, '_').squeeze('_')
   end
 
   def replacements
     booleans = { 'true' => '', 'false' => ' not' }
     booleans.each { |k, v| booleans[k] = "triangle is#{v} #{triangle}" }
   end
 
-  def do
-    "triangle.#{triangle}?"
+  def workload
+    "triangle = Triangle.new(#{sides})
+    #{assert_or_refute} triangle.#{triangle}?, #{failure_message}"
```

Comment:
This relies on the indentation of the test case generator being the same as the indentation of the template file, Which is not a problem now, but is rather fragile.
Have a look at how Transpose handles indenting: https://github.com/exercism/xruby/pull/466